### DISCUSSION TITLE
Parameter Null Checking & CallerArgumentExpression

### DISCUSSION THREAD

**Discussion by BOBJohnson** [dotnet/csharplang#6302] [Language Ideas] [answered]
I don't know if this is the correct place to post this. Apologies if it is not.

I love the idea of the Parameter Null Checking feature (https://github.com/dotnet/csharplang/blob/main/proposals/param-nullchecking.md) in the works.

I also like the fairly new CallerArgumentExpression and can provide some extremely useful information in tracking down problems, such as when an argument is null and it should not be.

I have no idea if this is even remotely feasible, but I would love for the new !! operator to be combined with CallerArgumentExpression attribute.

For example:

    void M(string name!!) {
        ...
    }

Would emit:

    void M(string name, [CallerArgumentExpression("name")] string? nameExpression = null) {
        if (name is null) {
            throw new ArgumentNullException(nameExpression);
        }
        ...
    }

Edit: fixed the missing string? from the method parameters.

**Comment by svick** [comment, marked as answer]
[That feature has been removed from C# 11](https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-04-13.md#parameter-null-checking) and is currently not planned to be included in a future version of C#.
